src/redprl/pretty.sml

```diff
@@ -1 +1,46 @@
 structure PP = PrettyPrint (AnsiColors)
+
+structure TermPrinter = 
+struct
+  structure Abt = RedPrlAbt
+  structure ShowVar = Abt.Var
+  structure ShowSym = Abt.Sym
+  structure O = RedPrlOpData
+  structure P = RedPrlParameterTerm
+
+  structure UP = UnparseAbt (structure Abt = Abt and Unparse = Unparse)
+
+  open Abt O Unparse
+
+  fun @@ (f, x) = f x
+  infix 0 @@ $ $$ \
+
+  val paramToString = P.toString ShowSym.toString
+
+  fun notation m = 
+    case Abt.out m of
+      (* , x *)
+      POLY (HYP_REF x) $ [] => SOME (atom @@ ", " ^ ShowVar.toString x)
+    | (* A -> B *)
+      MONO FUN $ [_ \ a, (_, []) \ b] =>
+        SOME (atom @@ toString a ^ " -> " ^ toString b)
```

Comment:
Here, we should use the infixing mechanism, so, it would be something like this:

```sml
infix' (Right, 3, "->") (unparse a, unparse b)
```